get-started/debug-and-profile.md

```diff
@@ -1,8 +1,174 @@
+
+For example,
+
+1. `[executor/compiler.go:Compile](https://github.com/pingcap/tidb/blob/5c95062cc34d6d37e2e921f9bddba6205b43ee3a/executor/compiler.go#L48)` is where every SQL is compiled and optimized
+2. `[planner/planner.go:Optimize](https://github.com/pingcap/tidb/blob/5c95062cc34d6d37e2e921f9bddba6205b43ee3a/planner/optimize.go#L80)` is where the SQL optimization starts
+3. `[executor/adapter.go:ExecStmt.Exec](https://github.com/pingcap/tidb/blob/5c95062cc34d6d37e2e921f9bddba6205b43ee3a/executor/adapter.go#L312)` is where the SQL plan turns into executor, and where the SQL execution starts
+4. Each `Open`, `Next` and `Close` function of each executor marks the volcano-style execution logic
+
+As you are reading the TiDB source code, you are strongly encouraged to set a breakpoint and use the debugger to trace the execution whenever you feel confused or uncertain about the code.
+
+## Using `pprof` for profiling
+
+For any database system, performance is always an important issue, and if you want to know where the performance bottleneck is, Go provides a powerful profiling tool called `pprof` for this purpose.
+
+### Gathering runtime profiling information through HTTP end points
+
+Normally, when you have a TiDB server running, it will expose a profiling end point through HTTP at `http://127.0.0.1:10080/debug/pprof/profile`, and you can get the profile result through
+
+```bash
+$ curl -G "127.0.0.1:10080/debug/pprof/profile?seconds=45" > profile.profile
+$ go tool pprof -http 127.0.0.1:4001 profile.profile
+```
+
+which will capture the profiling information for 45 seconds, and then provide a web view for the profiling result at `127.0.0.1:4001`, which contains the [flame graph](http://www.brendangregg.com/flamegraphs.html) of the execution and many more views that can help you diagnosis the performance bottleneck.
+
+Similarly, other runtime information can also be gathered through this end point, for example:
+
+- Goroutine:
+
+```bash
+curl -G "127.0.0.1:10080/debug/pprof/goroutine" > goroutine.profile
+```
+
+- Trace:
+
+```bash
+$ curl -G "127.0.0.1:10080/debug/pprof/trace?seconds=3" > trace.profile
+$ go tool trace -http 127.0.0.1:4001 trace.profile
+```
+
+- Heap:
+
+```bash
+$ curl -G "127.0.0.1:10080/debug/pprof/heap" > heap.profile
+$ go tool pprof -http 127.0.0.1:4001 heap.profile
+```
+
+You can refer to Go's [diagnostics document](https://golang.org/doc/diagnostics) for how these information can be analyzed.
+
+### Profiling during benchmarking
+
+When you are proposing performance related features for TiDB, it is recommended to also include a benchmark result to proof the performance gain or your code won't introduce any performance regression. In this case, you need to write your own benchmark test like in `executor/benchmark.go`.
+
+For example, if you want to benchmark the window functions, there are already `BenchmarkWindow` in the benchmark tests, so you can run
```

Comment:
```suggestion
For example, if you want to benchmark the window functions, because `BenchmarkWindow` are already in the benchmark tests, you can run the following commands to get the benchmark result.
```